src/yaml/from_yaml.cr

```diff
@@ -22,39 +22,87 @@ end
 
 def Nil.new(pull : YAML::PullParser)
   location = pull.location
-  value = pull.read_scalar
-  if value.empty?
+  value = pull.read_plain_scalar
+  if YAML::NULL_VALUES.includes? value
     nil
   else
-    raise YAML::ParseException.new("Expected nil, not #{value}", *location)
+    raise YAML::ParseException.new("Expected null, not '#{value}'", *location)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  location = pull.location
+  value = pull.read_plain_scalar
+  if YAML::TRUE_VALUES.includes? value
+    true
+  elsif YAML::FALSE_VALUES.includes? value
+    false
+  else
+    raise YAML::ParseException.new("Expected boolean, not '#{value}'", *location)
+  end
 end
 
 {% for type in %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) %}
   def {{type.id}}.new(pull : YAML::PullParser)
     location = pull.location
+    value = pull.read_plain_scalar
     begin
-      {{type.id}}.new(pull.read_scalar)
+      value.to_{{type.gsub(/^(U|I)I?nt/, "\\1").downcase.id}}(prefix: true, underscore: true)
     rescue ex
       raise YAML::ParseException.new(ex.message.not_nil!, *location)
     end
   end
 {% end %}
 
-def String.new(pull : YAML::PullParser)
-  pull.read_scalar
-end
+{% for type in %w(Float32 Float64) %}
```

Comment:
Same here: maybe use the bit size instead of FloatXX constants, and avoid raising/rescuing an exception, just to raise again:

```crystal
{% for bits in [32, 64] %}
  def Float{{bits}}
    location = pull.location
    value = pull.read_plain_scalar
    if float = value.to_f{{bits}}?
      float
    else
      raise YAML::ParseException.new("Invalid Float{{bits}} number", *location)
    end
  end
{% end %}
```